### ISSUE TITLE
Configure release mechanism through Travis.

### ISSUE THREAD

**Issue by sjoerdtalsma** [talsma-ict/context-propagation#4]
So the release is not dependent on the developer's laptop with SSH + GPG keys and passwords.
- [x] [introduce Travis CI build user](https://github.com/talsma-ict/enumerables/issues/11)
- [x] add maven wrapper
- [x] [configure travis build](#3)
- [x] configure release script
- [x] sort out credentials in travis

**Comment by sjoerdtalsma** [comment]
About maven central publishing (without storing secrets in git):

https://dracoblue.net/dev/uploading-snapshots-and-releases-to-maven-central-with-travis/

**Comment by sjoerdtalsma** [comment]
This depends on #3 to be resolved.
Then, necessary release credentials and travis+maven configuration must be added.

**Comment by sjoerdtalsma** [comment]
Travis job now builds master branch + pull requests.
Point of attention: The graphviz tool `dot` seems unavailable on vanilla travis-ci.
Look into a workaround or alternative way for this:

```
[WARNING] Javadoc Warnings
[WARNING] java.io.IOException: Cannot run program "/usr/bin/dot": error=2, No such file or directory
[WARNING] at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
[WARNING] at java.lang.Runtime.exec(Runtime.java:620)
[WARNING] at net.sourceforge.plantuml.cucadiagram.dot.ProcessRunner$MainThread.startThreads(ProcessRunner.java:157)
[WARNING] at net.sourceforge.plantuml.cucadiagram.dot.ProcessRunner$MainThread.runJob(ProcessRunner.java:117)
[WARNING] at net.sourceforge.plantuml.api.TimeoutExecutor$MyThread.run(TimeoutExecutor.java:74)
[WARNING] Caused by: java.io.IOException: error=2, No such file or directory
[WARNING] at java.lang.UNIXProcess.forkAndExec(Native Method)
[WARNING] at java.lang.UNIXProcess.<init>(UNIXProcess.java:247)
[WARNING] at java.lang.ProcessImpl.start(ProcessImpl.java:134)
[WARNING] at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
[WARNING] ... 4 more
[WARNING] java.io.IOException: Cannot run program "/usr/bin/dot": error=2, No such file or directory
[WARNING] at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
[WARNING] at java.lang.Runtime.exec(Runtime.java:620)
[WARNING] at net.sourceforge.plantuml.cucadiagram.dot.ProcessRunner$MainThread.startThreads(ProcessRunner.java:157)
[WARNING] at net.sourceforge.plantuml.cucadiagram.dot.ProcessRunner$MainThread.runJob(ProcessRunner.java:117)
[WARNING] at net.sourceforge.plantuml.api.TimeoutExecutor$MyThread.run(TimeoutExecutor.java:74)
[WARNING] Caused by: java.io.IOException: error=2, No such file or directory
[WARNING] at java.lang.UNIXProcess.forkAndExec(Native Method)
[WARNING] at java.lang.UNIXProcess.<init>(UNIXProcess.java:247)
[WARNING] at java.lang.ProcessImpl.start(ProcessImpl.java:134)
[WARNING] at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
[WARNING] ... 4 more
```

**Comment by sjoerdtalsma** [comment]
Graphviz support was added in https://github.com/talsma-ict/context-propagation/commit/b297a5996b451afe7e06373d18de579f63b1554b. Seems to work 👍

**Comment by sjoerdtalsma** [comment]
This depends on https://github.com/talsma-ict/enumerables/issues/11 to configure a dedicated Travis CI user.

**Comment by sjoerdtalsma** [comment]
I think all prerequisites have been fulfilled, let's give it a test by merging to master.

**Comment by sjoerdtalsma** [comment]
Added the following environment variables to Travis-CI:

- `GH_USER`
- `GH_TOKEN`
- `GPG_EXECUTABLE`
- `GPG_KEYNAME`
- `GPG_PASSPHRASE`
- `GPG_OWNERTRUST`
- `GPG_SECRETKEYS`
- `SONATYPE_USER`
- `SONATYPE_PASSWORD`

**Comment by sjoerdtalsma** [comment]
Unfortunately git-push still doesn't work. (token removed to be recreated later)

**Comment by sjoerdtalsma** [comment]
Added maven release plugin to add the correct tag: https://github.com/talsma-ict/context-propagation/commit/a538e8dcd427537a2bf3ebd5cc5b4668e637fb29

**Comment by sjoerdtalsma** [comment]
Issue can be closed, release 0.4.3 has been published from Travis without any (additional) manual intervention.
